conf/lets-encrypt-renew-certificate.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+web_service='nginx'
+config_file='/github-contributions/conf/lets-encrypt-config.ini'
+le_path='/opt/letsencrypt'
+exp_limit=30;
+
+if [ ! -f $config_file ]; then
+    echo "[ERROR] config file does not exist: $config_file"
+    exit 1;
+fi
+
+domain=githubcontributions.io
+cert_file="/etc/letsencrypt/live/$domain/fullchain.pem"
+
+if [ ! -f $cert_file ]; then
+    echo "[ERROR] certificate file not found for domain $domain."
+fi
+
+exp=$(date -d "`openssl x509 -in $cert_file -text -noout|grep "Not After"|cut -c 25-`" +%s)
+datenow=$(date -d "now" +%s)
+days_exp=$(echo \( $exp - $datenow \) / 86400 |bc)
+
+echo "Checking expiration date for $domain..."
+
+if [ "$days_exp" -gt "$exp_limit" ] ; then
+    echo "The certificate is up to date, no need for renewal ($days_exp days left)."
+    exit 0;
+else
+    echo "The certificate for $domain is about to expire soon. Starting webroot renewal script..."
+    $le_path/letsencrypt-auto certonly -a webroot --agree-tos --renew-by-default --config $config_file
+    echo "Reloading $web_service"
+    /usr/sbin/service $web_service reload
+    echo "Renewal process finished for domain $domain"
+    exit 0;
+fi
```

Comment:
Basically copied it from https://www.digitalocean.com/community/tutorials/how-to-secure-nginx-with-let-s-encrypt-on-ubuntu-14-04, but I'm happy to take credit.